### PR TITLE
Add external DNS chart for Azure

### PR DESCRIPTION
**Pull request by r7vme**
Add new chart for Azure "external-dns". External DNS needed just to add DNS entry for ingress load balancer, that's now managed by Kubernetes itself in Azure.

Part-of: https://github.com/giantswarm/giantswarm/issues/3268

### CODE REVIEW

**Comment by rossf7** [comment]
@r7vme This looks good and #142 is ready. Will merge once its approved.

File: pkg/v4/resource/chartconfig/desired.go [hunk 1]
```diff
@@ -45,6 +45,12 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		desiredChartConfigs = append(desiredChartConfigs, chartConfig)
 	}
 
+	// Only enable External DNS for Azure.
+	if r.provider == label.ProviderAzure {
+		chartConfig := newExternalDNSChartConfig(clusterConfig, r.projectName)
+		desiredChartConfigs = append(desiredChartConfigs, chartConfig)
+	}
```
Comment (xh3b4sd) [inline pkg/v4/resource/chartconfig/desired.go]: This is the wrong place to do this IMO. The resources in `pkg/` are actually meant for all providers. There are specific resources for Azure and we should deal with Azure specifics there.

Comment (r7vme) [reply pkg/v4/resource/chartconfig/desired.go]: Sorry, i'm not into cluster-operator. May be @rossf7 can answer, why we have this under `pkg`?

Comment (rossf7) [reply pkg/v4/resource/chartconfig/desired.go]: The logic for creating chartconfigs is the same for all providers. But we need to be able to install provider specific charts like in this case.

Other way to do this is to extract all logic to a service and have resources per provider with different desired states. That is doable but some code duplication. WDYT?

cc @fgimenez @indyfree

**Pull request merged**